mlrun/datastore/targets.py

```diff
@@ -1208,19 +1212,18 @@ def __init__(
         warehouse: str = None,
         table_name: str = None,
     ):
-        attrs = {
-            "url": url,
-            "user": user,
-            "database": database,
-            "schema": db_schema,
-            "warehouse": warehouse,
-            "table": table_name,
-        }
-        extended_attrs = {
-            key: value for key, value in attrs.items() if value is not None
-        }
-        attributes = {} if not attributes else attributes
-        attributes.update(extended_attrs)
+        attributes = attributes or {}
+        if url:
+            attrs = {
+                "url": url,
+                "user": user,
+                "database": database,
+                "db_schema": db_schema,
+                "warehouse": warehouse,
+                "table": table_name,
+            }
+            attributes.update(attrs)
```

Comment:
Hm, is this correct? If `url` is set we override all these attrs, otherwise we ignore the parameters? Wouldn't this make more sense?
```suggestion
        attributes = attributes or {}
        if url:
            attributes["url"] = url
        if user:
            attributes["user"] = user
        if database:
            attributes["database"] = database
        if db_schema:
            attributes["db_schema"] = db_schema
        if warehouse:
            attributes["warehouse"] = warehouse
        if table_name:
            attributes["table"] = table_name
```